test/runTests.ts

```diff
@@ -0,0 +1,57 @@
+import { readdirSync } from 'node:fs'
+import { run } from 'node:test'
+import process from 'node:process'
+import { spec as Spec } from 'node:test/reporters'
+import { basename } from 'node:path'
+import { cpus } from 'node:os'
+
+const spec = new Spec()
+
+let exitCode = 0
+
+const files = readdirSync(__dirname)
+	.filter((f) => f.endsWith('.ts') && f !== basename(__filename))
+	.map((f) => `${__dirname}/${f}`)
+
+const start = Date.now()
+
+const testStream = run({
+	files,
+	timeout: 60 * 1000,
+	concurrency: cpus().length,
+})
+
+testStream.compose(spec).pipe(process.stdout)
+
+const summary: string[] = []
+
+testStream.on('test:fail', (data) => {
+	exitCode = 1
+	const error = data.details.error
+
+	summary.push(
+		`${data.file} - "${data.name}" (${Math.round(
+			data.details.duration_ms,
+		)}ms)\n${error.toString()} `,
+	)
+})
+
+testStream.on('test:stderr', (data) => {
+	summary.push(`${data.file} - Error:\n${data.message} `)
+})
```

Comment:
Dunno why but the summary is missed when running tests using `run` function. It is printed only when running them from a single file. Maybe it's a bug or I dunno, I created my custom summary in this way, also this is needed to correct the exit code in case there are errors